kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,185 @@
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties;
+
+	@Test(expected = NullPointerException.class)
+	public void shouldFailWhenNull() {
+		givenTheFollowingProperties(null);
+		whenNetworkPropsIsCreated();
+	}
+
+	@Test
+	public void shouldReturnPropertiesWithGetProperties() {
+		givenTheMapWith("testKey1", "testString1");
+		whenNetworkPropsIsCreated();
+		thenCompareGetPropertiesToPassedProperties();
+	}
+
+	@Test
+	public void shouldReturnStringWithGet() {
+		givenTheMapWith("testKey1", "testString1");
+		whenNetworkPropsIsCreated();
+		thenGetReturnsExpectedObjectWithKey("testKey1");
```

Comment:
I find it hard to follow what is tested here. It's similar to what we had with the `given` stuff  in that I have to go at the bottom of the file to understand what is the expected object.

How about something like:

```suggestion
                whenGetIsCalledWith("testKey1", String.class);
                
                thenNoExceptionOccurred();
                thenTheResultIsNotNull();
                thenTheResultBelongsTo(String.class);
                thenTheResultIsEqualTo("testString1");
```

again: more readable, more flexible and easier to debug.

Please note that not all the method here make sense (the "belongtoclass" doesn't convince me), try to find out the ones that work for you.